File: app/src/main/java/com/mithril/mobilegoldenleaf/MobileGoldenLeafApplication.java
package com.mithril.mobilegoldenleaf;

import android.app.Application;

import androidx.room.Room;

import com.mithril.mobilegoldenleaf.database.MobileGoldenLeafDataBase;
import com.mithril.mobilegoldenleaf.database.dao.CategoryDao;
import com.mithril.mobilegoldenleaf.database.dao.ProductDao;
import com.mithril.mobilegoldenleaf.models.Category;
import com.mithril.mobilegoldenleaf.models.Product;

public class MobileGoldenLeafApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        //initializeDatabase();
    }

    private void initializeDatabase() {
        Category bebidas = new Category("Bebidas");
        //Category leite_substitutos = new Category("Leite / Substitutos");

        MobileGoldenLeafDataBase database = Room
                .databaseBuilder(this, MobileGoldenLeafDataBase.class, "GoldenLeafDataBase.db")
                .allowMainThreadQueries()
                .build();

        CategoryDao categoryDao = database.getCategoryDao();
        long id = categoryDao.save(bebidas);
        ProductDao productDao = database.getProductDao();
        productDao.save(new Product((int) id, "Coca Cola", "Refrigerante Fanta Uva 2L", "12345678", 5.25));
        //dao.save(new Product(2, "Qboa", "Água sanitária", "12345687", 3.50));

    }
}

File: app/src/main/java/com/mithril/mobilegoldenleaf/MobileGoldenLeafApplication.kt
package com.mithril.mobilegoldenleaf

import android.app.Application

import androidx.room.Room

import com.mithril.mobilegoldenleaf.persistence.MobileGoldenLeafDataBase
import com.mithril.mobilegoldenleaf.models.Category
import com.mithril.mobilegoldenleaf.models.Product
import java.math.BigDecimal

class MobileGoldenLeafApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        initializeDatabase()
    }

    private fun initializeDatabase() {
        val bebidas = Category("Bebidas")
        val leite_substitutos = Category("Leite / Substitutos")

        val database = Room
                .databaseBuilder(this, MobileGoldenLeafDataBase::class.java, "GoldenLeaf.db")
                .allowMainThreadQueries()
                .build()

        val categoryDao = database.categoryRepository
        val bebidasId = categoryDao.save(bebidas)
        val productRepository = database.productRepository
        productRepository.save(
                Product(bebidasId, "Coca Cola", "REFRIGERANTE COCA-COLA GARRAFA 2L", "7894900011517", BigDecimal(5.25)))
        val leite_substitutosId = categoryDao.save(leite_substitutos)
        productRepository.save(
                Product(leite_substitutosId, "NESTLE IDEAL", "NESTLÉ IDEAL SACHÊ", "7891000015391", BigDecimal(3.50)))

    }
}
